### DISCUSSION TITLE
$freeze/$once returning reactive value and preventing in-place updates of it

### DISCUSSION THREAD

**Discussion by smbdelse** [alpine-collective/alpine-magic-helpers#51] [Ideas] [answered]
Hi!

I'm wondering if it is possible to render value of some x-data expression or prop in it's current state for that moment, and keep it without updates. Probably it would be better to have it as modifier for x-bind prop. This would be helpful mostly for values based on iteration in x-for loops that cannot relay on simple index (i.e. to simplify logic in multiple nested x-for loops). Or maybe there already exist some good solution for that particular problem?

**Comment by KevinBatdorf** [comment, marked as answer]
You can store the current state easily enough with something like `$el.__x.getUnobservedData()` ([demo](https://codepen.io/KevinBatdorf/pen/KKgPRKB?editors=1000))

Can you expand on the example case a bit more? I'm not necessarily following it but interested to hear more.

**Reply by smbdelse** [reply]
I wasn't aware of above method, but I'm still unsure if it can be applicable for printing values. [Here is a simple example with use of global variable](https://codepen.io/KamilDyrek/pen/ExgYRKW?editors=1011)

**Reply by KevinBatdorf** [reply]
If you want a property to not be reactive, you can just define it in `x-init` after Alpine has already set up the reactive stuff. Like this:

https://codepen.io/KevinBatdorf/pen/f0b1b67a9c4feeddd019829fa83d4ef6?editors=1011

**Reply by smbdelse** [reply]
Thank you for those examples, they will definitely help me :)

**Reply by KevinBatdorf** [reply]
No problem :) There's also a Discord with people helping each other as well https://discord.gg/snmCYk3
